src/ape/pytest/contextmanagers.py

```diff
@@ -1,26 +1,118 @@
+        except ValueError as exc:
             raise AssertionError(
-                f"Transaction did not revert.\n"
-                f"However, an exception of type {type(exc_value)} occurred: {exc_value}"
-            ) from exc_value
+                f"Could not fetch contract at {receipt.receiver} to check dev message"
+            ) from exc
+
+        if contract.contract_type.sourcemap is None:
+            raise AssertionError("Could not fetch contract source to check dev message")
+
+        trace = deque(exc.trace)
+
+        if trace[-1].op != "REVERT":
+            raise AssertionError("Transaction trace did not end with REVERT")
+
+        pc = None
+
+        # To find a suitable line for inspecting dev messages, we must start at the revert and work
+        # our way backwards. If the last frame's PC is in the PC map, the offending line is very
+        # likely a 'raise' statement.
+        if str(trace[-1].pc) in contract.contract_type.pcmap:
+            pc = str(trace[-1].pc)
+
+        # Otherwise we must traverse the trace backwards until we find our first suitable candidate.
+        # It should be noted that when an error happens, typically the contract will execute:
+        #
+        #   <...>
+        #   JUMPI    <- jump to revert routine on a failing condition
+        #
+        #   <...>
+        #   JUMPDEST <- beginning of revert routine
+        #   PUSH1
+        #   DUP1
+        #   REVERT
+        #
+        # So we must step backwards until we find the initial instructions that caused the failing
+        # condition. This will not be the JUMPI or even an arithmetic instruction like DIV, but
+        # rather the starting instruction for a given line.
+        else:
+            while len(trace) > 0:
+                frame = trace.pop()
+                if str(frame.pc) in contract.contract_type.pcmap:
+                    pc = str(frame.pc)
+
+                    break
+
+        # We were unable to find a suitable PC that matched the compiler's map.
+        if not pc:
+            raise AssertionError("Could not find line that caused revert")
+
+        offending_line_num = contract.contract_type.pcmap[pc][0]
+
+        # The compiler PC map had PC information, but not source information.
+        if offending_line_num is None:
+            raise AssertionError("Could not find line that caused revert")
+
+        contract_source_lines = (
+            ape.project.extract_manifest()
+            .sources[contract.contract_type.source_id]
+            .content.split("\n")
+        )
 
-        actual = exc_value.revert_message
+        # Line numbers will not be zero-indexed
+        offending_line = contract_source_lines[offending_line_num - 1]
+
+        # Match '# dev: (...)' with any variation of whitespace
+        match = re.search(r"#\s*dev\s*:\s*([\w\d\s]+)$", offending_line)
+
+        if not match:
+            raise AssertionError(f"Could not find dev message '{self.dev_message}'")
```

Comment:
As noted in https://github.com/ApeWorX/ape/pull/1084/files#r1006224591 I think I may change this to actually print out the offending line (granted it's not too large...)